Airwallex/AirwallexPayment/Sources/CardProvider.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  CardProvider.swift
+//  AirwallexPayment
+//
+//  Created by Weiping Li on 20/8/25.
+//  Copyright © 2025 Airwallex. All rights reserved.
+//
+
+#if canImport(AirwallexCore)
+import AirwallexCore
+#endif
+import Foundation
+
+class CardProvider: PaymentProvider {
+    
+    override class func canHandle(_ session: AWXSession, paymentMethod: AWXPaymentMethodType) -> Bool {
+        guard session is Session else {
+            return false
+        }
+        do {
+            try AWXCardProvider.validateMethodTypeAndSession(paymentMethodType: paymentMethod, session: session)
+            return true
+        } catch {
+            return false
+        }
+    }
+    
+    func confirmIntentWithCard(_ card: AWXCard,
+                               billing: AWXPlaceDetails? = nil,
+                               saveCard: Bool) async {
+        debugLog("Start payment confirm. Type: Card. Intent Id: \(unifiedSession.paymentIntent.id)")
+        let method = AWXPaymentMethod()
+        method.type = AWXCardKey
+        method.billing = billing
+        method.card = card
+        method.customerId = unifiedSession.paymentIntent.customerId
+        
+        let consentOptions = if saveCard && unifiedSession.paymentIntent.customerId != nil {
+            PaymentConsentOptions(nextTriggeredBy: .customerType)
+        } else {
+            unifiedSession.paymentConsentOptions
+        }
+        
+        let request = createConfirmIntentRequest(
+            method: method,
+            consent: nil,
+            consentOptions: consentOptions
+        )
+        await confirmIntent(request)
+    }
+    
+    /// Confirms a payment intent using an existing payment consent
+    /// - Parameter consent: The payment consent to use for confirmation
+    func confirmIntentWithConsent(_ consent: AWXPaymentConsent) async {
+        
+        do {
+            let methodId = consent.paymentMethod?.id
+            var cvc = consent.paymentMethod?.card?.cvc
+            // Collect CVC if needed for PAN cards
+            if let card = consent.paymentMethod?.card,
+               card.numberType == "PAN" && (cvc ?? "").isEmpty {
+                cvc = try await collectCVC(for:consent)
+            }
+            if let methodId {
+                if let options = unifiedSession.paymentConsentOptions  {
+                    // Create consent & confirm payment with existing payment method
+                    let request = createRequestForConsentConversion(
```

Comment:
I wonder why it's called consent conversion here, any CIT <-> MIT conversions?

Reply:
I prefer `createPaymentRequestWithConsentCreation` to make it clear that we send payment request with the effect of consent creation at the same time.